packages/@blockprotocol/graph/src/stdlib/subgraph/builder.ts

```diff
@@ -0,0 +1,68 @@
+import { addEntitiesToSubgraphByMutation } from "../../internal/mutate-subgraph.js";
+import {
+  Entity,
+  EntityEditionId,
+  GraphResolveDepths,
+  Subgraph,
+} from "../../types.js";
+
+/**
+ * Builds a Subgraph from a given set of entities, which may include 'link entities' –
+ * i.e. entities that represent relationships between other entities – or other entities.
+ *
+ * The set of entities should represent the result of a query on a graph,
+ * and the 'roots' and 'depths' for that query provided along with the data.
+ *
+ * Unbounded queries can currently only be represented via very high values for depths.
+ *
+ * This function does NOT verify that the provided depths are accurate for the data.
+ *   – the caller is responsible for this.
+ * It DOES check that the provided roots are present in the data.
+ *
+ * @param data – the data to build the subgraph from (which becomes the vertices)
+ * @param data.entities – the entities to build the subgraph from
+ * @param depths – the depth values to provide in the returned subgraph
+ * @param roots – the root values to provide in the returned subgraph
+ *
+ * @returns a Subgraph containing:
+ *   - 'vertices' containing the provided entities
+ *   - 'edges' calculated by the function, representing connections between vertices
+ *   - 'depths' as provided by the caller
+ *   - 'roots' as provided by the caller
+ *
+ * @throws if the provided roots are not present in the data
+ *
+ * @todo 1. add support for ontology vertices (e.g. entity types)
+ * @todo 2. support null values for depths, for the result of queries with unlimited depths.
```

Comment:
Why do we want to support this? I can't really imagine situations where this would be a reasonable expectation of an EA, and except in _**very very very**_ specific edge-cases you should be expecting exponential growth in the size of a query (where honestly even a depth of > 10 might be too much to handle in memory).

I think we should get blocks to make follow-up requests for data if they really do need more than a depth of 255.

Reply:
To be honest when I added this I was thinking about `mock-block-dock`, which uses this to represent its entire datastore as a subgraph. Happy to just leave it as `255` max and we can revisit if there's any ever reason to change it

Reply:
I think lets go with 255 until reason to not do so. (If only the 255 thing was _**easily**_ captured within JS/TS like it is in languages with differently sized number types)

Reply:
https://github.com/blockprotocol/blockprotocol/pull/869/commits/74ed5c6af3aa1339a460d6af1b62c6b49435ef1c